accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/authorizableinstaller/impl/AuthInstallerUserManagerPrefetchingImpl.java

```diff
@@ -91,7 +93,15 @@ public AuthInstallerUserManagerPrefetchingImpl(UserManager delegate, final Value
                 Group memberOfGroup = declaredMemberOf.next();
                 String memberOfGroupId = memberOfGroup.getID();
                 isMemberOfByAuthorizableId.get(authId).add(memberOfGroupId);
-                nonRegularUserMembersByAuthorizableId.get(memberOfGroupId).add(authId);
+                final Set<String> nonRegularUserMembers;
+                if (!nonRegularUserMembersByAuthorizableId.containsKey(memberOfGroupId)) {
+                    // in case the group was not found by the previous query
+                    nonRegularUserMembers = new HashSet<String>();
+                    nonRegularUserMembersByAuthorizableId.put(memberOfGroupId, nonRegularUserMembers);
+                } else {
+                    nonRegularUserMembers = nonRegularUserMembersByAuthorizableId.get(memberOfGroupId);
+                }
+                nonRegularUserMembers.add(authId);
```

Comment:
as this code is self contained, may be it will be easier to read if we separate it in a new method

```
    private void addMemberToAuthId(final String memberOfGroupId, final String authId) {
        final Set<String> nonRegularUserMembers;
        if (!nonRegularUserMembersByAuthorizableId.containsKey(memberOfGroupId)) {
            // in case the group was not found by the previous query
            nonRegularUserMembers = new HashSet<String>();
            nonRegularUserMembersByAuthorizableId.put(memberOfGroupId, nonRegularUserMembers);
        } else {
            nonRegularUserMembers = nonRegularUserMembersByAuthorizableId.get(memberOfGroupId);
        }
        nonRegularUserMembers.add(authId);
    }
```

and then we can call it here like 

```addMemberToAuthId(memberOfGroupId, authId);```